File: app/src/main/java/me/echeung/moemoekyun/ui/view/SquareLayout.java
package me.echeung.moemoekyun.ui.view;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;

@SuppressLint("AppCompatCustomView")
public class SquareLayout extends FrameLayout {

    public SquareLayout(Context context) {
        super(context);
    }

    public SquareLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SquareLayout(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);

        int dimen = Math.min(getMeasuredWidth(), getMeasuredHeight());
        setMeasuredDimension(dimen, dimen);
    }

}

File: app/src/main/kotlin/me/echeung/moemoekyun/ui/view/SquareLayout.kt
package me.echeung.moemoekyun.ui.view

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout
import kotlin.math.min

@SuppressLint("AppCompatCustomView")
class SquareLayout : FrameLayout {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)

        val dimen = min(measuredWidth, measuredHeight)
        setMeasuredDimension(dimen, dimen)
    }
}
